pkg/api/types.go

```diff
@@ -193,6 +193,8 @@ type VolumeSource struct {
 	Secret *SecretVolumeSource `json:"secret"`
 	// NFS represents an NFS mount on the host that shares a pod's lifetime
 	NFS *NFSVolumeSource `json:"nfs"`
+	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
+	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
```

Comment:
You'll also want to add this to PersistentVolumeSource to make it a provisionable resource.

If Gluster is not to be exposed to the end user (only the admin provisions it, users claim it), then after the PV framework is fully merged you can remove GFS from VolumeSource and leave it in PVS.  This hides it completely from the pod author.